src/utils/object.js

```diff
@@ -44,3 +44,26 @@ export function map(opt_initial) {
 export function hasOwn(obj, key) {
   return hasOwn_.call(obj, key);
 }
+
+/**
+ * Returns the value of the property referenced by dot-separated keys.
+ * e.g.
+ * `getPath({a: {b: [{c: 2}]}}, 'a.b[0].c') === 2`
+ *
+ * @param {T} obj a map-like value
```

Comment:
Can you help me understand why is this different than `hasOwn`, which returns a boolean but specifies a [`@param {T}  obj`](https://github.com/cvializ/amphtml/blob/cd8a5f9fe0f624d975c90b439e0c76b0b4960118/src/utils/object.js#L39)?